node/middlewares/auth.ts

```diff
@@ -0,0 +1,53 @@
+import { AuthenticationError } from '@vtex/api'
+
+export async function auth(ctx: Context, next: () => Promise<void>) {
+  const {
+    header,
+    clients: { vtexId, sphinx },
+    state,
+  } = ctx
+
+  const appkey = header['x-vtex-api-appkey'] as string | undefined
+  const apptoken = header['x-vtex-api-apptoken'] as string | undefined
+  const authCookie = header.vtexidclientautcookie as string | undefined
+
+  if (authCookie) {
+    const authenticatedUser = await vtexId.getAuthenticatedUser(authCookie)
+
+    // When the auth cookie is invalid, the API returns null
+    if (authenticatedUser) {
+      const isAdmin = await sphinx.isAdmin(authenticatedUser.user)
+
+      const { user, userId } = authenticatedUser
+
+      state.userProfile = {
+        userId,
+        email: user,
+        role: isAdmin ? 'admin' : 'store-user',
+      }
+
+      if (isAdmin) {
```

Comment:
At this point, shouldn't we also validate the type of access that admin user or key has in case of access without specific roles? Perhaps the televandas/admin owner role is the only one that should have access to perform this type of action.